protostar/migrator/migrator_contract_identifier_resolver.py

```diff
@@ -15,18 +30,15 @@ def __init__(
 
     def resolve(self, contract_identifier: str) -> Path:
         if "." in contract_identifier:
-            return Path(contract_identifier)
-        return self._compile_contract_by_contract_name(contract_identifier)
+            contract_path = Path(contract_identifier)
+            return checked_contract_path(contract_path)
 
-    def _compile_contract_by_contract_name(self, contract_name: str) -> Path:
-        contract_class = self._project_compiler.compile_contract_from_contract_name(
-            contract_name
-        )
-        output_file_path = (
-            CompiledContractWriter(contract=contract_class, contract_name=contract_name)
-            .save(
-                output_dir=self._migrator_datetime_state.get_compilation_output_path()
-            )
-            .compiled_contract_path
+        return self._get_built_contract_by_name(contract_identifier)
+
+    def _get_built_contract_by_name(self, contract_name: str) -> Path:
+        compilation_output_dir = self._project_compiler.get_compilation_output_dir(
+            Path(BuildCommand.COMPILATION_OUTPUT_ARG.default)
```

Comment:
[Functionality]: This code will break if user will use name for the output directory. I think the migrate command should take extra argument instead.

Reply:
done